test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1144,17 +1142,15 @@ private synchronized void reset(boolean wipeData) throws IOException {
         final List<Settings> settings = new ArrayList<>();
 
         for (int i = 0; i < numSharedDedicatedMasterNodes; i++) {
```

Comment:
I wonder if these loops that setup each node's settings can be moved to pluggable methods on the node configuration source, so that the node roles used can be replaced altogether, instead of hacking it into the existing "data" node?